src/lib.rs

```diff
@@ -404,16 +400,341 @@
 //! #   ClientLoginFinishParameters::default(),
 //! # )?;
 //! let server_login_finish_result = server_login_start_result.state.finish(
-//!    client_login_finish_result.message,
+//!     client_login_finish_result.message,
 //! )?;
+//!
 //! assert_eq!(
 //!    client_login_finish_result.shared_secret,
 //!    server_login_finish_result.shared_secret,
 //! );
 //! # Ok::<(), ProtocolError>(())
 //! ```
 //! If the protocol completes successfully, then the server obtains a `server_login_finish_result.shared_secret` which is guaranteed to
-//! match `client_login_finish_result.shared_secret`. Otherwise, on failure, the [ServerLogin::finish] algorithm outputs the error [InvalidLoginError](errors::PakeError::InvalidLoginError).
+//! match `client_login_finish_result.shared_secret` (see the [Shared Secret](#shared-secret) section).
+//! Otherwise, on failure, the [ServerLogin::finish] algorithm outputs the error [InvalidLoginError](errors::PakeError::InvalidLoginError).
+//!
+//! # Advanced Usage
+//!
+//! This implementation offers support for several optional features of OPAQUE, described below. They are not critical to the
+//! execution of the main protocol, but can provide additional security benefits which can be suitable for various applications that rely on
+//! OPAQUE for authentication.
+//!
+//! ## Shared Secret
+//!
+//! Upon a successful completion of the OPAQUE protocol (the client runs login with the same password used during registration),
+//! the client and server have access to a shared secret, which is a pseudorandomly distributed
+//! 32-byte string which only the client and server know. Multiple login runs using the same password for the same client will
+//! produce different shared secrets, distributed as uniformly random strings. Thus, the shared secret can be used as a session secret
+//! for a secure channel between the client and server.
+//!
+//! The shared secret can be accessed from the `shared_secret` field of [ClientLoginFinishResult] and [ServerLoginFinishResult]. See
+//! the combination of [Client Login Finish](#client-login-finish) and [Server Login Finish](#server-login-finish) for example usage.
+//!
+//! ## Checking Server Consistency
+//!
+//! A [ClientLoginFinishResult] contains the `server_s_pk` field, which represents the static public key of the server that is established
+//! during the setup phase. This can be used by the client to verify the authenticity of the server it engages with during the login phase. In particular,
```

Comment:
nit: something like : "contains the `server_s_pk`, which is extracted from ..." or "As this value is extracted from the envelope generated at registration ..."

Reply:
Technically with the new envelope structure, it is actually not extracted from the envelope. It instead must be supplied in order to verify the mac on the envelope, but the value itself isn't present in the envelope.

Reply:
:eyes: Thanks!